"HW9"

rickdemonn opened the pull request.

Commit:
Revert "HW9"

mfarsikov commented on src/main/java/hillel/spring/reviews/ReviewService.java:
```diff
@@ -0,0 +1,128 @@
+
+        petRepo.findById(review.getPetId()).orElseThrow(PetNotFoundException::new);
+
+        review.setDate(LocalDateTime.now(ZoneId.systemDefault()));
```
please use a Clock

mfarsikov commented on src/main/java/hillel/spring/reviews/ReviewService.java:
```diff
@@ -0,0 +1,128 @@
+    private final ReviewRepo reviewRepo;
+    private final ReviewConfig reviewConfig;
+    private final AppointmentRepo appointmentRepo;
+    private ReviewReport reviewReport;
```
**It is a really bad idea** to have a mutable state in your components.
`ReviewReport` should be created on the fly, and not kept in the service state.
Also `ReviewReport` is not a `@Component` it is just a value object (from Domain Driven Design).

mfarsikov commented on src/main/java/hillel/spring/reviews/ReviewService.java:
```diff
@@ -0,0 +1,128 @@
+        return reviewRepo.findById(id);
+    }
+
+    public void makeReview(Review review) throws RuntimeException {
```
remove `throws` declaration

mfarsikov commented on src/main/java/hillel/spring/reviews/Review.java:
```diff
@@ -0,0 +1,53 @@
+    private String equipment;
+    private String specialistQualification;
+    private String effectivenessOfTheTreatment;
+    private String overallRating;
```
Can't all the ratings be described using numbers?

mfarsikov commented on src/main/java/hillel/spring/reviews/ReviewService.java:
```diff
@@ -0,0 +1,128 @@
+    private final PetRepo petRepo;
+    private final ReviewRepo reviewRepo;
+    private final ReviewConfig reviewConfig;
+    private final AppointmentRepo appointmentRepo;
```
it is better to avoid accessing other repositories from your service. Please rework this, and replace these repositories with appropriate services.

mfarsikov commented on src/main/resources/application.yml:
```diff
@@ -22,4 +22,11 @@ doctor.specializations:
+      - "**"
+      - "***"
+      - "****"
+      - "*****"
```
:(

rickdemonn committed:
Home Work 10 (#11)

* Home Work 10

*Add Pagination
*Add Validation

* little fix size of List<specializations>

Commit:
home work 9 fix

mfarsikov commented on src/main/java/hillel/spring/reviews/ReviewService.java:
```diff
@@ -0,0 +1,110 @@
+
+        petRepo.findById(review.getPetId()).orElseThrow(PetNotFoundException::new);
+
+        review.setDate(LocalDateTime.now(Clock.systemUTC()));
```
You can define a bean of type `Clock`. This allows using fixed clock for tests

mfarsikov commented on src/main/java/hillel/spring/reviews/ReviewService.java:
```diff
@@ -0,0 +1,110 @@
+@Service
+@AllArgsConstructor
+public class ReviewService {
+    private final PetRepo petRepo;
```
ReviewService has access to PetRepo :(

Commit:
home work 9 second fix

rickdemonn committed:
home work 11 (#12)

* home work 11

*Add dependence to info service to create and update doctor
*add optimistic lock to update doctor
*refactoring test and add WireMock

* home work 11 fix
